web/src/narrow_state.ts

```diff
@@ -33,7 +33,10 @@ export function filter(): Filter | undefined {
 
 export function search_terms(): Term[] {
     if (current_filter === undefined) {
-        return new Filter(page_params.narrow).terms();
+        if (page_params.narrow !== undefined) {
+            return new Filter(page_params.narrow).terms();
+        }
+        return [];
     }
     return current_filter.terms();
```

Comment:
Similarly can you check how the call from `search.js` into this function in `narrow_state` will handle the change:
```
export function search_string(): string {                                                               
    return Filter.unparse(search_terms());                                                              
}                                                                                                       
```

Reply:
OK this is indeed a bug; starting a search from the Inbox or Recent view gives you this as the initial operators:

![image](https://github.com/zulip/zulip/assets/2746074/58473a04-f02e-494c-9c61-2e4685fb5aeb)

I think the fix is that this should just be `[]`.